cli/internal/fs/turbo_json.go

```diff
@@ -107,14 +116,53 @@ func readTurboJSON(path turbopath.AbsolutePath) (*TurboJSON, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	err = jsonc.Unmarshal(data, &turboJSON)
+
 	if err != nil {
 		println("error unmarshalling", err.Error())
 		return nil, err
 	}
+
 	return turboJSON, nil
 }
 
+// UnmarshalJSON deserializes TurboJSON objects into struct
+func (c *TurboJSON) UnmarshalJSON(data []byte) error {
+	raw := &rawTurboJSON{}
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+
+	envVarDependencies := make(util.Set)
+	globalFileDependencies := make(util.Set)
+
+	for _, value := range raw.GlobalEnv {
+		if strings.HasPrefix(value, envPipelineDelimiter) {
+			log.Printf("[WARNING] Ignoring \"%s\" environment variable declaration. Remove %s prefix.", value, envPipelineDelimiter)
```

Comment:
will remove this as discussed in https://github.com/vercel/turborepo/pull/1932#discussion_r971108639